snapshot/pkg/controller/cache/actual_state_of_world.go

```diff
@@ -70,7 +70,7 @@ func (asw *actualStateOfWorld) AddSnapshot(snapshot *crdv1.VolumeSnapshot) error
 	asw.Lock()
 	defer asw.Unlock()
 
-	snapshotName := MakeSnapshotName(snapshot.Metadata.Namespace, snapshot.Metadata.Name)
+	snapshotName := MakeSnapshotName(snapshot)
```

Comment:
this could add the same snapshot multiple times (since they have different UIDs)

Reply:
If they have different UID they would be different snapshots (possible if the old is being deleted and new one being created?).  There can still only be one snapshot object for a given name+namespace in etcd